src/main/java/de/sakpaas/backend/service/LocationService.java

```diff
@@ -273,4 +287,19 @@ private void createNewLocation(OsmResultLocationListDto.OsmResultLocationDto osm
     );
     this.save(location);
   }
+
+  /**
+   * Deletes the given Location and all depending entities.
+   *
+   * @param location Location that needs to be deleted
+   */
+  private void delete(Location location) {
+    if (location == null) {
+      return;
+    }
```

Comment:
Ich bin mir nicht einig ob es schön ist, wenn wir hier einen null check machen. Meiner Meinung nach sollte der Benutzer von `delete` dafür zuständig sein, dass `location` nicht `null` ist.
Was sagt der Rest von @SAKPaaS/backend dazu?

Reply:
Ist auch besser finde ich. Sonst hat der "Benutzer" der die Methode aufruft keine Möglichkeit zu erfahren, ob der Delete-Vorgang erfolgreich war.

Reply:
Ich habe den `null`-Check in die aufrufende Methode verschoben.